### ISSUE TITLE
No extension metadata for mapped superclass

### ISSUE THREAD

**Issue by Miljar** [Atlantic18/DoctrineExtensions#1696]
I have a single table inheritance setup, where my base entity has the Timestampable fields mapped like this:
```xml
<?xml version="1.0" encoding="UTF-8"?>

<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping"
                  xmlns:gedmo="http://gediminasm.org/schemas/orm/doctrine-extensions-mapping"
                  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                  xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping
                                      http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">

    <entity name="MyProject\Common\Entity\Page\AbstractPage"
        inheritance-type="SINGLE_TABLE"                                          
        repository-class="MyProject\Common\Repository\PageRepository"
        table="page">

        <id name="id" column="id" type="integer">
            <generator strategy="AUTO" />
        </id>

        <discriminator-column name="type" type="pagetype" length="25" />
        <discriminator-map>
            <discriminator-mapping value="rootpage" class="MyProject\Common\Entity\Page\RootPage" />
            <discriminator-mapping value="contentpage" class="MyProject\Common\Entity\Page\ContentPage" />
        </discriminator-map>

        <!-- TIMESTAMPABLE FIELDS -->
        <field name="createdAt" column="created_at" type="datetime">
            <gedmo:timestampable on="create"/>
        </field>
        <field name="updatedAt" column="updated_at" type="datetime">
            <gedmo:timestampable on="update"/>
        </field>
        <!-- /TIMESTAMPABLE FIELDS -->
    </entity>

</doctrine-mapping>
```

When trying to persist a child class, the timestampable fields are left empty.

Example child class mapping:

```xml
<?xml version="1.0" encoding="UTF-8"?>

<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping"
                  xmlns:gedmo="http://gediminasm.org/schemas/orm/doctrine-extensions-mapping"
                  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                  xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping
                                      http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">

    <mapped-superclass name="MyProject\Common\Entity\Page\RootPage">
    </mapped-superclass>

</doctrine-mapping>
```

It appears there is a check in the ExtensionMetadataFactory which blocks retrieving the config if the object is a mapped superclass: https://github.com/Atlantic18/DoctrineExtensions/blob/v2.4.x/lib/Gedmo/Mapping/ExtensionMetadataFactory.php#L74

When I remove that check, everything seems to be working. Is there any reason why this check needs to be here?

**Comment by l3pp4rd** [comment]
Well, I never imagined that it was possible to use MappedSuperclasses other than extending it. Seems like inheritance mapping allows to use it in discriminator map. You are the first probably to use it this way without actually extending it. You can submit a pull request for this change to 2.4.x branch, did not thought doctrine allows such combination

**Comment by Miljar** [comment]
As it turns out, my mapping was "wrong". My base class cannot be a mapped superclass, Doctrine doesn't allow inheritance mapping there. My leaf classes were mapped superclasses, but that was actually a mistake by me. When I converted them to entities instead of mapped superclasses, everything worked fine.

I did actually look at the source code. Removing the mentioned check caused some unit tests for Tree to error. Since everything can be solved by adjusting my mapping, I suggest we defer or ignore this issue.
